diff --git a/src/fashn_vton/tryon_mmdit.py b/src/fashn_vton/tryon_mmdit.py
index fab0ac3..0a7a0cc 100644
--- a/src/fashn_vton/tryon_mmdit.py
+++ b/src/fashn_vton/tryon_mmdit.py
@@ -32,7 +32,7 @@ def attention(q: Tensor, k: Tensor, v: Tensor, pe: Tensor) -> Tensor:
 
 def rope(pos: Tensor, dim: int, theta: int) -> Tensor:
     assert dim % 2 == 0
-    scale = torch.arange(0, dim, 2, dtype=torch.float64, device=pos.device) / dim
+    scale = torch.arange(0, dim, 2, dtype=(torch.float32 if pos.device.type == "mps" else torch.float64), device=pos.device) / dim
     omega = 1.0 / (theta**scale)
     out = torch.einsum("...n,d->...nd", pos, omega)
     out = torch.stack([torch.cos(out), -torch.sin(out), torch.sin(out), torch.cos(out)], dim=-1)
